servicetests/tests/FleetProvisioning/src/main/java/fleetProvisioning/FleetProvisioning.java

```diff
@@ -0,0 +1,252 @@
+import software.amazon.awssdk.crt.mqtt.MqttClientConnection;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnectionEvents;
+import software.amazon.awssdk.crt.mqtt.QualityOfService;
+import software.amazon.awssdk.crt.mqtt5.Mqtt5ClientOptions;
+import software.amazon.awssdk.crt.mqtt5.Mqtt5Client;
+import software.amazon.awssdk.crt.mqtt5.packets.ConnectPacket;
+import software.amazon.awssdk.iot.AwsIotMqttConnectionBuilder;
+import software.amazon.awssdk.iot.AwsIotMqtt5ClientBuilder;
+import software.amazon.awssdk.iot.iotidentity.IotIdentityClient;
+import software.amazon.awssdk.iot.iotidentity.model.CreateKeysAndCertificateRequest;
+import software.amazon.awssdk.iot.iotidentity.model.CreateKeysAndCertificateResponse;
+import software.amazon.awssdk.iot.iotidentity.model.CreateKeysAndCertificateSubscriptionRequest;
+import software.amazon.awssdk.iot.iotidentity.model.ErrorResponse;
+import software.amazon.awssdk.iot.iotidentity.model.RegisterThingRequest;
+import software.amazon.awssdk.iot.iotidentity.model.RegisterThingResponse;
+import software.amazon.awssdk.iot.iotidentity.model.RegisterThingSubscriptionRequest;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.util.HashMap;
+import com.google.gson.Gson;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import utils.commandlineutils.CommandLineUtils;
+import utils.mqttclientconnectionwrapper.*;
+import ServiceTestLifecycleEvents.ServiceTestLifecycleEvents;
+
+public class FleetProvisioning {
+
+    static CompletableFuture<Void> gotResponse;
+    static IotIdentityClient iotIdentityClient;
+
+    static CreateKeysAndCertificateResponse createKeysAndCertificateResponse = null;
+    static RegisterThingResponse registerThingResponse = null;
+
+    static long responseWaitTimeMs = 5000L; // 5 seconds
+
+    static MqttClientConnectionWrapper createConnection(CommandLineUtils.SampleCommandLineData cmdData, Integer mqttVersion) {
+        if (mqttVersion == 3) {
+            try (AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder
+                    .newMtlsBuilderFromPath(cmdData.input_cert, cmdData.input_key)) {
+                builder.withClientId(cmdData.input_clientId)
+                    .withEndpoint(cmdData.input_endpoint)
+                    .withPort((short)cmdData.input_port)
+                    .withCleanSession(true)
+                    .withProtocolOperationTimeoutMs(60000);
+                Mqtt3ClientConnectionWrapper connWrapper = new Mqtt3ClientConnectionWrapper();
+                connWrapper.connection = builder.build();
+                if (connWrapper.connection == null) {
+                    throw new RuntimeException("MQTT311 connection creation failed!");
+                }
+                return connWrapper;
+            } catch (Exception ex) {
+                throw new RuntimeException("Failed to create MQTT311 connection", ex);
```

Comment:
Question: would the python script reported error if we throw an exception?

Reply:
Probably, the script won't report it itself, but the exception will be reported by Java. I threw an exception from main for experiment:
```
Error:  Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.1.0:java (default-cli) on project FleetProvisioningServiceTest: An exception occurred while executing the Java class. Invalid MQTT version specified: 3 -> [Help 1]
```